src/v/kafka/server/replicated_partition.h

```diff
@@ -48,10 +55,21 @@ class replicated_partition final : public kafka::partition_proxy::impl {
     }
 
     model::offset high_watermark() const final {
+        if (
+          _partition->is_read_replica_mode_enabled()
+          && _partition->cloud_data_available()) {
+            return _partition->last_cloud_offset();
+        }
         return _translator->from_log_offset(_partition->high_watermark());
     }
 
     model::offset last_stable_offset() const final {
+        if (
+          _partition->is_read_replica_mode_enabled()
+          && _partition->cloud_data_available()) {
+            // There is no difference between HWM and LO in this mode
+            return _partition->last_cloud_offset();
```

Comment:
The same https://github.com/redpanda-data/redpanda/pull/5242/commits/86a82f22bd3de73e90e06690c6e89fb1e31dcfbb#r909568900 comment applies here